tests/5.0/teams_loop/test_target_teams_loop_defaultmap.c

```diff
@@ -0,0 +1,249 @@
+  }
+#pragma omp target teams loop defaultmap(tofrom)
+  for (int i = 0; i < N; i++) {
+    host_data[i] += device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(from)
+  clause with target teams loop construct.
+*/
+int testDefaultMapFrom() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(from)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(to)
+  clause with target teams loop construct.
+*/
+int testDefaultMapTo() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(to) map(tofrom: host_data)
+  for (int i = 0; i < N; i++) {
+    host_data[i] += device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(default)
+  clause with target teams loop construct.
+*/
+int testDefaultMapDefault() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(default)
+  for (int i = 0; i < N; i++) {
+    host_data[i] += device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(firstprivate)
+  clause with target teams loop construct.
+*/
+int testDefaultMapFirstPrivate() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(firstprivate)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != 0);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom:aggregate)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFromAggr() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(tofrom:aggregate) map(from: host_data)
```

Comment:
defaultmap will not map host_data "to" the device since there is a explicit time.